clang/lib/Basic/Targets.cpp

```diff
@@ -685,9 +685,15 @@ std::unique_ptr<TargetInfo> AllocateTarget(const llvm::Triple &Triple,
       switch (HT.getEnvironment()) {
       default: // Assume MSVC for unknown environments
       case llvm::Triple::MSVC:
-        assert(HT.getArch() == llvm::Triple::x86_64 &&
-               "Unsupported host architecture");
-        return std::make_unique<MicrosoftX86_64_SPIR64TargetInfo>(Triple, Opts);
+        if (HT.getArch() == llvm::Triple::aarch64)
+          return std::make_unique<MicrosoftARM64_SPIR64TargetInfo>(Triple,
+                                                                   Opts);
+        else if (HT.getArch() == llvm::Triple::x86_64)
+          return std::make_unique<MicrosoftX86_64_SPIR64TargetInfo>(Triple,
+                                                                    Opts);
+        else
+          llvm::report_fatal_error(
+              "Unsupported host architecture (not x86_64 or aarch64)");
```

Comment:
Nit; Just to make the [coding style](https://llvm.org/docs/CodingStandards.html#don-t-use-else-after-a-return) gods happy:
```suggestion
        if (HT.getArch() == llvm::Triple::aarch64)
          return std::make_unique<MicrosoftARM64_SPIR64TargetInfo>(Triple,
                                                                   Opts);
        if (HT.getArch() == llvm::Triple::x86_64)
          return std::make_unique<MicrosoftX86_64_SPIR64TargetInfo>(Triple,
                                                                    Opts);
        llvm::report_fatal_error(
            "Unsupported host architecture (not x86_64 or aarch64)");
```
It could also be a switch, but I don't have a preference there.

Reply:
Thanks @steffenlarsen ! Have updated.